vm/builtin/io.cpp

```diff
@@ -727,7 +727,7 @@ namespace rubinius {
     int res = ::select(fd+1, &set, 0, 0, &tv);
 
     if(res == 0) {
-      Exception::errno_eagain_error(state, "no data ready");
+      Exception::errno_wait_readable(state, EAGAIN);
```

Comment:
This is a bit of a hack because `errno` is not set when `select` returns 0. Without explicitly passing `EAGAIN` we'd have to handle this case in `Exception::errno_wait_readable` which I think is the wrong place to do so (since this behaviour is very specific to `select`).